code/src/App.js

```diff
@@ -1,9 +1,88 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { HappyThought } from 'components/HappyThought'
+import { PostThought } from 'components/PostThought'
+import { Loader } from 'components/Loader'
+import { API_URL, LIKES_API_URL } from 'utils/urls'
 
 export const App = () => {
+  const [thoughts, setThoughts] = useState([])
+  const [newThought, setNewThought] = useState('')
+  const [loading, setLoading] = useState(true)
+  const [typing, setTyping] = useState(false)
+
+  useEffect(() => {
+    refreshThoughts()
+  }, [])
+
+  const refreshThoughts = () => {
+    setLoading(true)
+    fetch(API_URL)
+      .then(res => res.json())
+      .then(data => {
+        setThoughts(data)
+      })
+      .finally(() => setLoading(false))
+  }
+
+  const handleTyping = thinking => {
+    setTyping(true)
+    setNewThought(thinking)
+  }
+
+  const handleFormSubmit = e => {
+    e.preventDefault()
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ message: newThought }),
+    }
+
+    fetch(API_URL, options)
+      .then(res => res.json())
+      .then(data => {
+        refreshThoughts()
+        // setThoughts([data, ...thoughts])
```

Comment:
Remember to delete commented out code (if you are not planning to use it later on ofc) :)